komga/src/main/kotlin/org/gotson/komga/domain/service/SeriesLifecycle.kt

```diff
@@ -197,17 +197,23 @@ class SeriesLifecycle(
   fun getThumbnail(seriesId: String): ThumbnailSeries? {
     val selected = thumbnailsSeriesRepository.findSelectedBySeriesIdOrNull(seriesId)
 
-    if (selected == null || !selected.exists()) {
+    if (selected == null || (selected.type == ThumbnailSeries.Type.SIDECAR && !selected.exists())) {
       thumbnailsHouseKeeping(seriesId)
       return thumbnailsSeriesRepository.findSelectedBySeriesIdOrNull(seriesId)
     }
 
     return selected
   }
 
+  fun getThumbnailBytesById(seriesId: String, thumbnailId: String): ByteArray? =
+    thumbnailsSeriesRepository.findOneOrNull(seriesId, thumbnailId)?.thumbnail
+
   fun getThumbnailBytes(seriesId: String, userId: String): ByteArray? {
     getThumbnail(seriesId)?.let {
-      return File(it.url.toURI()).readBytes()
+      return when (it.type) {
+        ThumbnailSeries.Type.USER_UPLOADED -> it.thumbnail
+        ThumbnailSeries.Type.SIDECAR -> it.url?.toURI()?.let { uri -> File(uri).readBytes() }
+      }
```

Comment:
You could do something similar to `BookLifecycle`:
```kotlin
return when {
        it.thumbnail != null -> it.thumbnail
        it.url != null -> File(it.url.toURI()).readBytes()
        else -> null
      }
```
We don't really care about the type, but more about whether the payload is in the DB or on file